src/wallet/test/psbt_wallet_tests.cpp

```diff
@@ -64,7 +64,7 @@ BOOST_AUTO_TEST_CASE(psbt_updater_test)
+    BOOST_CHECK_EQUAL(final_hex, "70736274ff01009a020000000258e87a21b56daf0c23be8e7070456c336f7cbaa5c8757924f545887bb2abdd750000000000ffffffff838d0427d0ec650a68aa46bb0b098aea4422c071b2ca78352a077959d07cea1d0100000000ffffffff0270aaf00800000000160014d85c2b71d0060b09c9886aeb815e50991dda124d00e1f5050000000016001400aea9a2e5f0f876a588df5546e8742d1d87008f00000000000100bb0200000001aad73931018bd25f84ae400b68848be09db706eac2ac18298babee71ab656f8b0000000048473044022058f6fc7c6a33e1b31548d481c826c015bd30135aad42cd67790dab66d2ad243b02204a1ced2604c6735b6393e5b41691dd78b00f0c5942fb9f751856faa938157dba01feffffff0280f0fa020000000017a9140fb9463421696b82c833af241c78c17ddbde493487d0f20a270100000017a91429ca74f8a08f81999428185c97b5d852e4063f6187650000000104475221029583bf39ae0a609747ad199addd634fa6108559d6c5cd39b4c2183f1ab96e07f2102dab61ff49a14db6a7d02b0cd1fbb78fc4b18312b5b4e54dae4dba2fbfef536d752ae2206029583bf39ae0a609747ad199addd634fa6108559d6c5cd39b4c2183f1ab96e07f10d90c6a4f000000800300008000000080220602dab61ff49a14db6a7d02b0cd1fbb78fc4b18312b5b4e54dae4dba2fbfef536d710d90c6a4f0000008003000080010000800000220203a9a4c37f5996d3aa25dbac6b570af0650394492942460b354753ed9eeca5877110d90c6a4f000000800300008004000080002202027f6399757d2eff55a136ad02c684b1838b6556e5f1b6b34282a94b6b5005109610d90c6a4f00000080030000800500008000");
```

Comment:
This is still signed with keys derived from `m/0'/0'/<0-3>'` instead of `m/0'/3'/<0-3>'`. This is caused by the call under the "FillPSBT" comment - on line 56. The transaction inside that psbt is, decoded into json format:

```
{
    "txid": "82efd652d7ab1197f01a5f4d9a30cb4c68bb79ab6fec58dfa1bf112291d1617b",
    "hash": "82efd652d7ab1197f01a5f4d9a30cb4c68bb79ab6fec58dfa1bf112291d1617b",
    "version": 2,
    "locktime": 0,
    "vin": [
        {
            "txid": "75ddabb27b8845f5247975c8a5ba7c6f336c4570708ebe230caf6db5217ae858",
            "vout": 0,
            "scriptSig": {
                "asm": "",
                "hex": ""
            },
            "sequence": 4294967295
        }, 
        {
            "txid": "1dea7cd05979072a3578cab271c02244ea8a090bbb46aa680a65ecd027048d83",
            "vout": 1,
            "scriptSig": {
                "asm": "",
                "hex": ""
            },
            "sequence": 4294967295
        }
    ],
    "vout": [
        {
            "value": 1.4999,
            "n": 0,
            "scriptPubKey": {
                "asm": "0 d85c2b71d0060b09c9886aeb815e50991dda124d",
                "hex": "0014d85c2b71d0060b09c9886aeb815e50991dda124d",
                "type": "witness_v0_keyhash"
            }
        }, 
        {
            "value": 1.00,
            "n": 1,
            "scriptPubKey": {
                "asm": "0 00aea9a2e5f0f876a588df5546e8742d1d87008f",
                "hex": "001400aea9a2e5f0f876a588df5546e8742d1d87008f",
                "type": "witness_v0_keyhash"
            }
        }
    ]
}
``` 

Also note that the outputs are segwit outputs.

Reply:
Thank you.  I am getting this JSON output after decoding hex parameter [https://bpa.st/23RQ](FillPSBT decoded on bpaste).  I'll look into it but, honestly, I don't quite get it yet. 😕

Reply:
We can fix that under the item "segwit and protocol dependencies" in the tracker. This test is only using the wallet derivation code, it is not testing it. So we're not accumulating technical debt by not fixing this now, we're just not reducing it here.